modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -581,6 +601,45 @@ public Object getWorkOwner() {
     }
   }
 
+  private class WithCompletionCallback implements RetryCallback {
+
+    private final CompletionCallback completionCallback;
+    private final RetryCallback delegate;
+
+    WithCompletionCallback(RetryCallback delegate, CompletionCallback completionCallback) {
+      this.delegate = delegate;
+      this.completionCallback = completionCallback;
+    }
+
+    @Override
+    public void doWork(RetryContext context) throws Exception {
+      try {
+        delegate.doWork(context);
+        this.completionCallback.success();
+      } catch (Exception ex) {
+        this.completionCallback.error(ex);
+        throw ex;
+      }
+    }
+
+    @Override
+    public String getWorkDescription() {
+      return delegate.getWorkDescription();
+    }
+
+    @Override
+    public Object getWorkOwner() {
+      return delegate.getWorkOwner();
+    }
+  }
+
+  private interface CompletionCallback {
```

Comment:
we have a ton of interfaces like this already. Simply reuse `ReactiveReconnectionCallback`

Reply:
Done